sql/core/src/test/scala/org/apache/spark/sql/DatasetPrimitiveSuite.scala

```diff
@@ -223,16 +223,6 @@ class DatasetPrimitiveSuite extends QueryTest with SharedSparkSession {
     checkDataset(Seq(Queue(true)).toDS(), Queue(true))
     checkDataset(Seq(Queue("test")).toDS(), Queue("test"))
     checkDataset(Seq(Queue(Tuple1(1))).toDS(), Queue(Tuple1(1)))
-
-    checkDataset(Seq(ArrayBuffer(1)).toDS(), ArrayBuffer(1))
```

Comment:
For complex reasons, I don't think these tests / use cases are possible as-is in Scala 2.13. They are niche, so I jsut removed them.

Reply:
Although this means the removal of test coverage in Scala 2.12, I'm +1 for now. We can add back later after we finished everything in Scala 2.13.